logbook-spring-boot-autoconfigure/src/main/java/org/zalando/logbook/autoconfigure/LogbookAutoConfiguration.java

```diff
@@ -345,11 +345,20 @@ public FilterRegistrationBean logbookFilter(final Logbook logbook) {
                     .withFormRequestMode(properties.getFilter().getFormRequestMode());
             return newFilter(filter, FILTER_NAME, Ordered.LOWEST_PRECEDENCE);
         }
+        
+        private static FilterRegistrationBean newFilter(final Filter filter, final String filterName, final int order) {
+            @SuppressWarnings("unchecked") // as of Spring Boot 2.x
+            final FilterRegistrationBean registration = new FilterRegistrationBean(filter);
+            registration.setName(filterName);
+            registration.setDispatcherTypes(REQUEST, ASYNC);
+            registration.setOrder(order);
+            return registration;
+        }
 
     }
 
     @Configuration(proxyBeanMethods = false)
-    @ConditionalOnClass(SecurityFilterChain.class)
+    @ConditionalOnClass({ SecurityFilterChain.class, Filter.class })
     @ConditionalOnWebApplication
```

Comment:
Maybe `@ConditionalOnWebApplication(type = SERVLET)` to be more explicit, like it's done in https://github.com/zalando/logbook/pull/863

Reply:
Yes, that's a good idea.

Reply:
I have added the web application type `SERVLET` to  both the `SecurityServletFilterConfiguration` and `ServletFilterConfiguration`.

Now I am not sure if keeping the condition `@ConditionalOnClass(Filter.class)` makes sense for these.